### PR TITLE
introduce party-has-responsiblity

### PR DESCRIPTION
**Pull request by wandmagic**
# Committer Notes

Introducing party has responsiblity constraint

### All Submissions:

- [ ] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                      xsi:schemaLocation="http://csrc.nist.gov/ns/oscal/1.0 https://github.com/usnistgov/OSCAL/releases/download/v1.1.2/oscal_ssp_schema.xsd"
+                      uuid="12345678-1234-4321-8765-123456789012">
+  <metadata>
+    <title>Enhanced Example System Security Plan</title>
+    <published>2024-08-01T14:30:00Z</published>
+    <last-modified>2024-08-01T14:30:00Z</last-modified>
+    <version>1.1</version>
+    <oscal-version>1.0.0</oscal-version>
+    <document-id scheme="https://example.com/identifiers">SSP-2024-002</document-id>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml:12]: Is this part of the demonstration on #700? If so, the point of the discussion around "just testing the constraint" means removing the content outside the scope of the test, so you can remove this bit, if that is the intent.

```suggestion
```

File: src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    </role>
+    <role id="asset-owner">
+      <title>Asset Owner</title>
+    </role>
+    <location uuid="11111112-0000-4000-9001-000000000009">
+      <address >
+        <country>WRONG</country>
+      </address>
+      <prop name='data-center' value='dc-zone-1' class='tertiary' ns="https://fedramp.gov/ns/oscal"/>
+    </location>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml:31]: Is this part of the demonstration on #700? If so, the point of the discussion around "just testing the constraint" means removing the content outside the scope of the test, so you can remove this bit, if that is the intent.

```suggestion
```

File: src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    <responsible-party role-id="creator">
+      <party-uuid>11111111-0000-4000-9000-000000000002</party-uuid>
+    </responsible-party>
+    <responsible-party role-id="developer">
+      <party-uuid>11111111-0000-4000-9000-000000000002</party-uuid>
+    </responsible-party>
+    
+    <remarks>
+      <p>This SSP is an example for demonstration purposes.</p>
+    </remarks>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml:53]: Is this part of the demonstration on #700? If so, the point of the discussion around "just testing the constraint" means removing the content outside the scope of the test, so you can remove this bit, if that is the intent.

```suggestion
```

File: src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  <back-matter>
+    <resource uuid="eeeeeeee-0000-4000-9000-00000000000e">
+      <!-- There is no <title/> to purposefully to trigger an error for negative testing. -->
+      <description>
+        <p>Detailed access control policy document</p>
+      </description>
+      <prop name="type" value="unsupported-type" ns="https://fedramp.gov/ns/oscal"/>
+      <!-- There is no <rlink/> to purposefully to trigger an error for negative testing. -->
+    </resource>
+  </back-matter>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml:207]: Is this part of the demonstration on #700? If so, the point of the discussion around "just testing the constraint" means removing the content outside the scope of the test, so you can remove this bit, if that is the intent.

```suggestion
```

**Review by aj-stein-gsa — COMMENTED** [review]
I don't want to request changes quite yet, I just want to make sure I understand the purpose of the `src/validations/constraints/content/ssp-party-has-one-responsibility-INVALID.xml` file in advance of #700. Lemme know.

**Comment by wandmagic** [comment]
moved to #718 with rebased branch

**Pull request closed without merging**